### PR TITLE
gh-95273: Normalise sqlite3 reference wording

### LINKED ISSUE #95273: Improve the sqlite3 reference
Improve the sqlite3 reference, using [Diataxis as a guideline](https://diataxis.fr/reference/). A lot of work has already been done here, with great help from Alex Waygood and CAM Gerlach. PRs so far[^1] have focused on...:

- [Be accurate](https://diataxis.fr/reference/#be-accurate)
- [Do nothing but describe](https://diataxis.fr/reference/#do-nothing-but-describe); making the wording more to the point

<details>
<summary>Previous work over the last few years</summary>

Both of these have been improved greatly by the following issues and PRs, amongst others:
- gh-94630
- gh-95235
- gh-95235
- gh-94940
- gh-94321
- gh-94628
- gh-94017
- gh-93890
- gh-93925
- gh-87260
- gh-89018
- gh-92780
- gh-90923
- gh-30660
- gh-29326
- gh-29200
- gh-29219
- gh-28238
- gh-26220
- gh-25003
- gh-23159
- gh-23965
- consistency: gh-31772
- consistency: gh-24746

</details>

There are remaining issues, though:

- [x] Be accurate:
    - #95379
    - #95523
    - #95530
- [x] [Be consistent](https://diataxis.fr/reference/#be-consistent): Most of the reference docs for sqlite3 functions and methods use the imperative mood. Some of the methods use other wordings. Suggesting to normalise, and use the imperative mood across the reference section.
  Improved with:
    - #95274 
    - #95525
    - Remove duplicate links
- [x] [Respect the structure of the machinery](https://diataxis.fr/reference/#respect-the-structure-of-the-machinery): Arrange the sections of reference to follow the architecture of the module. This is pretty good already, but I believe we may benefit from some adjustments, in particular:
    - [x] Consistently group attributes and methods
    - #95626
    - #95430
    - Align wording in function and method docstrings with documentation
- [x] [Do nothing but describe](https://diataxis.fr/reference/#do-nothing-but-describe): Avoid digressions and discussions, or even better, extract them to separate sections. Remove and or simplify examples.
    - #95351
    - #95383
    - gh-95840

[^1]: There have been too many PRs to list; some of them were done before Diataxis was even discussed.

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-95273

### CODE REVIEW

File: Doc/library/sqlite3.rst [hunk 1]
```diff
@@ -766,31 +768,28 @@ Connection Objects
...
 
    .. method:: iterdump
 
-      Returns an iterator to dump the database in an SQL text format.  Useful when
-      saving an in-memory database for later restoration.  This function provides
-      the same capabilities as the :kbd:`.dump` command in the :program:`sqlite3`
-      shell.
+      Return an :term:`iterator` to dump the database as SQL source code.
+      Useful when saving an in-memory database for later restoration.
+      Similar to the :kbd:`.dump` command in the :program:`sqlite3` shell.
```
Comment (erlend-aasland) [inline Doc/library/sqlite3.rst]: The original wording is wrong; `iterdump` does not handle all corner-cases, so it's better to say "similar" than "the same".

File: Doc/library/sqlite3.rst [hunk 2]
```diff
@@ -1003,20 +1004,19 @@ Cursor Objects
 
    .. method:: fetchone()
 
-      Fetches the next row of a query result set, returning a single sequence,
-      or :const:`None` when no more data is available.
+      Fetch the next row of a query result set as a :class:`tuple`.
```
Comment (erlend-aasland) [inline Doc/library/sqlite3.rst:1007]: We _could_ also mention that the returned row may be produced by a custom `row_factory`.

Comment (erlend-aasland) [reply Doc/library/sqlite3.rst:1007]: Ditto for fetchmany and fetchall.

File: Doc/library/sqlite3.rst [hunk 3]
```diff
@@ -745,10 +748,9 @@ Connection Objects
 
    .. attribute:: row_factory
 
-      You can change this attribute to a callable that accepts the cursor and the
-      original row as a tuple and will return the real result row.  This way, you can
-      implement more advanced ways of returning results, such  as returning an object
-      that can also access columns by name.
+      A callable that accepts two arguments,
+      a :class:`Cursor` object and the raw row results as a :class:`tuple`,
+      and return any object that fit the application programmers need.
```
Comment (CAM-Gerlach) [inline Doc/library/sqlite3.rst]: I'm rather confused by what

> return any object that fit the application programmers need.

is supposed to mean; aside from the grammar issues, it is unclear to me what I as a user am supposed to do with this.

File: Doc/library/sqlite3.rst [hunk 4]
```diff
@@ -766,31 +768,28 @@ Connection Objects
 
    .. attribute:: text_factory
 
-      Using this attribute you can control what objects are returned for the ``TEXT``
-      data type. By default, this attribute is set to :class:`str` and the
-      :mod:`sqlite3` module will return :class:`str` objects for ``TEXT``.
-      If you want to return :class:`bytes` instead, you can set it to :class:`bytes`.
+      A callable that accept a :class:`bytes` parameter and return a text
```
Comment (CAM-Gerlach) [inline Doc/library/sqlite3.rst]:
```suggestion
      A callable that accepts a :class:`bytes` parameter and returns a text
```

Grammar (singular/plural agreement)

File: Doc/library/sqlite3.rst (hunk 1 above)
Comment (CAM-Gerlach) [inline Doc/library/sqlite3.rst]: ``.dump`` is not, as far as I'm aware, a keyboard shortcut. If its a GUI element, then it should be

```suggestion
      Similar to the :guilabel:`.dump` command in the :program:`sqlite3` shell.
```

or if its a CLI command, then just the regular

```suggestion
      Similar to the ``.dump`` command in the :program:`sqlite3` shell.
```

File: Doc/library/sqlite3.rst [hunk 5]
```diff
@@ -1003,20 +1004,19 @@ Cursor Objects
...
+      Fetch the next row of a query result set as a :class:`tuple`.
+      Return :const:`None` when no more data is available.
 
 
    .. method:: fetchmany(size=cursor.arraysize)
 
-      Fetches the next set of rows of a query result, returning a list.  An empty
-      list is returned when no more rows are available.
+      Fetch the next set of rows of a query result as a :class:`list`.
+      Return an empty list is when no more rows are available.
```
Comment (CAM-Gerlach) [inline Doc/library/sqlite3.rst]:
```suggestion
      Return an empty list when no more rows are available.
```

Fix typo

**Review by CAM-Gerlach — COMMENTED** [review]
Looks good, just a few minor comments

File: Doc/library/sqlite3.rst (hunk 3 above)
Comment (erlend-aasland) [reply Doc/library/sqlite3.rst]: Well. I also find it hard to word this. The row factory can return whatever the programmer wishes. For one application, returning a string may be the right thing. For another application, returning a dataclass, or a complex object, or just a named tuple may be right.

File: Doc/library/sqlite3.rst (hunk 1 above)
Comment (erlend-aasland) [reply Doc/library/sqlite3.rst]: It's a CLI command, so let's just use double backticks.

File: Doc/library/sqlite3.rst (hunk 3 above)
Comment (erlend-aasland) [reply Doc/library/sqlite3.rst]: Perhaps just: _"return a row object tailored to the application specific need"_

Comment (AlexWaygood) [reply Doc/library/sqlite3.rst]: `and returns an arbitrary object representing a SQLite row`?

Comment (erlend-aasland) [reply Doc/library/sqlite3.rst]: Hm, "arbitrary" sounds ... arbitrary :) How about "returns a custom object", or "application specific object"?

Comment (erlend-aasland) [reply Doc/library/sqlite3.rst]: BTW, the "representing an SQLite row" suffix is definitely an improvement. 🐿️

Comment (AlexWaygood) [reply Doc/library/sqlite3.rst]: > Hm, "arbitrary" sounds ... arbitrary :) How about "returns a custom object", or "application specific object"?

"Custom object representing an SQLite row" works well imo, or you could just go with "object representing an SQLite row"

> 🐿️

Dammit, will I ever remember 😄

Comment (erlend-aasland) [reply Doc/library/sqlite3.rst]: Great, I'll land this using the former variant. Thank you so much, both of you :)

File: Doc/library/sqlite3.rst [hunk 6]
```diff
@@ -161,12 +161,13 @@ Module functions and constants
 
 .. data:: sqlite_version
 
-   The version number of the run-time SQLite library, as a string.
+   Version number of the run-time SQLite library as a :class:`string <str>`.
```
Comment (AlexWaygood) [inline Doc/library/sqlite3.rst]:
```suggestion
   Version number of the runtime SQLite library as a :class:`string <str>`.
```

File: Doc/library/sqlite3.rst [hunk 7]
```diff
@@ -161,12 +161,13 @@ Module functions and constants
...
 
 
 .. data:: sqlite_version_info
 
-   The version number of the run-time SQLite library, as a tuple of integers.
+   Version number of the run-time SQLite library as a :class:`tuple` of
```
Comment (AlexWaygood) [inline Doc/library/sqlite3.rst]:
```suggestion
   Version number of the runtime SQLite library as a :class:`tuple` of
```

File: Doc/library/sqlite3.rst [hunk 8]
```diff
@@ -1003,20 +1004,19 @@ Cursor Objects
...
+      Fetch the next row of a query result set as a :class:`tuple`.
+      Return :const:`None` when no more data is available.
 
 
    .. method:: fetchmany(size=cursor.arraysize)
 
-      Fetches the next set of rows of a query result, returning a list.  An empty
-      list is returned when no more rows are available.
+      Fetch the next set of rows of a query result as a :class:`list`.
+      Return an empty list when no more rows are available.
```
Comment (AlexWaygood) [inline Doc/library/sqlite3.rst]:
```suggestion
      Return an empty list if no more rows are available.
```

File: Doc/library/sqlite3.rst [hunk 9]
```diff
@@ -1003,20 +1004,19 @@ Cursor Objects
...
 
       The number of rows to fetch per call is specified by the *size* parameter.
-      If it is not given, the cursor's arraysize determines the number of rows
-      to be fetched. The method should try to fetch as many rows as indicated by
-      the size parameter. If this is not possible due to the specified number of
-      rows not being available, fewer rows may be returned.
+      If *size* is not given, :attr:`arraysize` determines the number of rows
+      to be fetched.
+      If fewer than *size* rows are available, fewer rows are returned.
```
Comment (AlexWaygood) [inline Doc/library/sqlite3.rst]: I'm not sure if this is accurate, but maybe:

```suggestion
      If fewer than *size* rows are available, as many rows as are available are returned.
```

File: Doc/library/sqlite3.rst [hunk 10]
```diff
@@ -1025,9 +1025,10 @@ Cursor Objects
 
    .. method:: fetchall()
 
-      Fetches all (remaining) rows of a query result, returning a list.  Note that
-      the cursor's arraysize attribute can affect the performance of this operation.
-      An empty list is returned when no rows are available.
+      Fetch all (remaining) rows of a query result as a :class:`list`.
+      Return an empty list when no rows are available.
```
Comment (AlexWaygood) [inline Doc/library/sqlite3.rst]:
```suggestion
      Return an empty list if no rows are available.
```

**Review by AlexWaygood — COMMENTED** [review]
Looks good; a few small points

File: Doc/library/sqlite3.rst [hunk 11]
```diff
@@ -1003,20 +1004,20 @@ Cursor Objects
 
    .. method:: fetchone()
 
-      Fetches the next row of a query result set, returning a single sequence,
-      or :const:`None` when no more data is available.
+      Fetch the next row of a query result set as a :class:`tuple`.
+      Return :const:`None` when no more data is available.
```
Comment (AlexWaygood) [inline Doc/library/sqlite3.rst]: Oops I missed one

```suggestion
      Return :const:`None` if no more data is available.
```

**Review by AlexWaygood — APPROVED** [review]
Nice!

**Comment by erlend-aasland** [comment]
Great, thanks again, both of you!

**Pull request merged**